app/main.py

```diff
@@ -1,6 +1,20 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Callable:
+        key = (args, tuple(kwargs.items()))
+
+        if key in cache_dict:
+            print("Getting from cache")
+            return cache_dict[key]
+        else:
+            print("Calculating new result")
+            cache_dict[key] = func(*args, **kwargs)
+            return cache_dict[key]
```

Comment:
it is better not to use the “else” method here

Reply:
agree, return cache_dict[key] once at the end of the function to avoid calling return twice